docs/eventing/sugar/README.md

```diff
@@ -0,0 +1,334 @@
+
+If the Broker of the given name already exists in the Namespace, the Sugar
+Controller will do nothing.
+
+## Automatic Trigger Creation
+
+The Sugar Controller can also watch for any resource that conforms to the duck
+type "Addressable", see [the defintion](TODO: add a link for addressable
+definition) or the [Discovery](TODO: link to the discovery documentation)
+ClusterDuckType
+[addressables.duck.knative.dev](https://github.com/knative-sandbox/discovery/blob/master/config/knative/addressables.duck.knative.dev.yaml).
+
+For any addressable that is found, the Sugar Controller will react to resources
+that have been labeled with the special label
+`eventing.knative.dev/autotrigger=enabled`.
+
+To produce one or more triggers automatically, provide the annotation:
+
+```
+annotations:
+  autotrigger.eventing.knative.dev/filter.attributes: |
+    [{"type":"cloudevents.event.type", "custom-attribute":"match-value"}]
+```
+
+`autotrigger.eventing.knative.dev/filter.attributes` is a JSON array of any map
+of string to string. Any valid CloudEvents attribute name and value are allowed.
+For each map in the array, a new Trigger will be created.
+
+Triggers produced by the Sugar Controller automatically create owner-refs to the
+original Addressable resource. When the labeled resource is deleted, the
+triggers will also be deleted.
+
+When the `autotrigger.eventing.knative.dev/filter.attributes` is modified, the
+Sugar Controller will attempt to realize the new config, which may result in
+deleting the triggers created for the old configuration.
+
+For resources that are owned by another Addressable resource, the Sugar
+Controller will attempt to not duplicate triggers if the labels and annotations
+were reflected down to the owned objects.
+
+The default Broker is used unless the broker annotation is specified:
+
+```
+annotations:
+  autotrigger.eventing.knative.dev/broker: hello-sugar
+```
+
+It is not supported to use more than one broker name with AutoTrigger.
+
+The Sugar Controller uses the label to enable or disable the AutoTrigger
+feature, and the annotations to provide the configuration. In this way, you are
+able to enable/disable the feature without editing the configuration.
+
+The Sugar Controller does not clean up Triggers if the
+`eventing.knative.dev/autotrigger` label is removed, and Trigger filters will no
+longer be updated.
+
+### Simple AutoTrigger Example
+
+When the following Knative Service is created:
+
+```yaml
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: hello-sugar
+  namespace: candyland
+  labels:
+    eventing.knative.dev/autotrigger: enabled
+  annotations:
+    trigger.eventing.knative.dev/filter.attributes: |
+      [{}]
+spec:
+  template:
+    spec:
+      containers:
+        - image: <some image>
+```
+
+The Sugar Controller will notice AutoTrigger is enabled for that resource, and
+produce a trigger subscribing "hello-sugar" to the Knative Service, the cluster
```

Comment:
s/produce/create/
subscribing "hello-sugar" Knative Service.
Or, I'm not sure what the "hello-sugar" to the Knative Service means.